### ISSUE TITLE
Improve demo bundling

### ISSUE THREAD

**Issue by JeroenVinke** [aurelia-ui-toolkits/aurelia-kendoui-plugin#188]
We can minimize the bundling config, and also push (most) of the `jspm_packages` directory to the gh-pages branch. Hopefully it will reduce the amount of errors you can get during bundling

**Comment by adriatic** [comment]
Perhaps it might make sense to present the issues we have in the context of updating the **[demo-kendo](https://github.com/aurelia-ui-toolkits/demo-kendo)** from the **[embedded sample app](https://github.com/aurelia-ui-toolkits/aurelia-kendoui-plugin/tree/master/sample)** to Guy; I would guess that our development setup would be encountered at many places.

**Comment by JeroenVinke** [comment]
[This comment](https://github.com/systemjs/builder/issues/166#issuecomment-100873615) is not visible enough IMO. If we publish (a large part) of jspm_packages, then files we forget to bundle are still served.

**Comment by adriatic** [comment]
> Ideally we do just want to have jspm_packages served in production though as it is also useful for dynamic loading scenarios.

Can you please explain this?

**Comment by JeroenVinke** [comment]
Sure.

So by bundling we mean concatenating (and minifying) all files in the jspm_packages folder, into a single file (lets say `app-bundle.js`). Instead of the browser having to fetch **every** file in the jspm_packages folder (in hundreds of different requests), the browser just has to download one file: `app-bundle.js`. This means that (if all works well) you don't have to include the `jspm_packages` folder when deploying, you just need to deploy the `app-bundle.js`.

Bundling is done by JSPM, but you have to tell JSPM which files to include in the bundle. It does some magic by looking up all files used by your application, but it misses a lot. That's why we have the [bundles.json](https://github.com/aurelia-ui-toolkits/demo-kendo/blob/master/build/bundles.json) file.

But what happens if we forget to include a file in this bundle configuration? Then the file won't be included in the `app-bundle.js`. As we (at the moment) don't have the `jspm_packages` folder in our `gh-pages` branch, it won't be able to download the missing file from there either. So, we get a 404.

It may be better to do this the other way around. We copy the entire jspm_packages folder, but exclude the files/folders we have included in the bundle configuration, and are thus in the bundle. This could reduce the amount of issues we have, as I stated in the paragraph above, because if we forget to bundle a file, JSPM can always get it from the `jspm_packages` folder.

**Comment by JeroenVinke** [comment]
So, I eventually got it working, but reverted this a few minutes ago. The jspm_packages folder contains 4500 files and is about 50MB, so you don't really want to push that to Github for every minor change. So, I excluded some folders, which brought the number of files down significantly, but increased the complexity of the bundling config. I don't think we can deal with this complexity, as it's hard enough to get the normal skeleton bundled, let alone a heavily customized bundle config, so I reverted the change.

I have improved the bundle config however, by removing all references to versions. So if we update dependences, we won't have to update the bundle config, or fix 404's if we forget ;)
